### PR TITLE
Remove ROS_WS env variable from foxy-ci which is causing the overlay to be installed on top of the underlay

### PR DESCRIPTION
**Pull request by JafarAbdi**
### Description

As you can see in https://travis-ci.com/github/ros-planning/moveit_task_constructor/jobs/489509921 having `ROS_WS` is causing the overlay ws to be installed on top of the underlay hence compiling the underlay again

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit/blob/master/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

**Review by rhaschke — APPROVED** [review]
Yeah, we don't define ROS_WS for MoveIt 1 as well.

File: .docker/ci-testing/Dockerfile [hunk 1]
```diff
@@ -16,21 +16,3 @@ RUN apt-get -qq update && \
...
-    find -L . -name COLCON_IGNORE -printf "%h\0" | xargs -0 rm -rf && \
-    apt-get -qq update && \
-    rosdep update -q && \
-    rosdep install -q -y --from-paths . --ignore-src --rosdistro ${ROS_DISTRO} --as-root=apt:false && \
-    # Clear apt-cache to reduce image size
-    rm -rf /var/lib/apt/lists/* && \
-    #
-    # Remove the source code from this container
-    cd $ROS_WS && \
-    rm -rf src/
```
Comment (rhaschke) [inline .docker/ci-testing/Dockerfile:36]: Implementing the TODO and removing all this is unrelated to this PR and should be a separate commit.

Comment (JafarAbdi) [reply .docker/ci-testing/Dockerfile:36]: These lines use `ROS_WS` this's why I combined the changes in the same PR

Comment (JafarAbdi) [reply .docker/ci-testing/Dockerfile:36]: I separated the changes into different commits, I don't think we need to implement anything here since we already run the commands in https://github.com/ros-planning/moveit2/blob/main/.docker/ci/Dockerfile

Comment (rhaschke) [reply .docker/ci-testing/Dockerfile:36]: In the past, it was important to fetch all dependencies from source here as well. Now, that everything has settled, we can drop this again (as stated in the TODO).

**Review by rhaschke — APPROVED** [review]
